component/src/main/java/org/wso2/extension/siddhi/gpl/execution/streamingml/classification/hoeffdingtree/util/AdaptiveHoeffdingTreeModel.java

```diff
@@ -0,0 +1,240 @@
+        } else {
+            hoeffdingAdaptiveTree.splitCriterionOption
+                    .setValueViaCLIString("GiniSplitCriterion");
+        }
+        hoeffdingAdaptiveTree.splitConfidenceOption.setValue(allowableSplitError);
+        hoeffdingAdaptiveTree.tieThresholdOption.setValue(breakTieThreshold);
+        hoeffdingAdaptiveTree.binarySplitsOption.setValue(binarySplitOption);
+        hoeffdingAdaptiveTree.noPrePruneOption.setValue(disablePrePruning);
+        hoeffdingAdaptiveTree.leafpredictionOption.setChosenIndex(leafpredictionStrategy);
+
+    }
+
+    /**
+     * @param cepEvent   event data
+     * @param classLabel class  label of the cepEvent
+     */
+    public void trainOnEvent(double[] cepEvent, String classLabel) {
+        cepEvent[noOfFeatures - 1] = addClass(classLabel);
+        Instance trainInstance = createMOAInstance(cepEvent);
+        trainInstance.setClassValue(cepEvent[noOfFeatures - 1]);
+        //training on the event instance
+        hoeffdingAdaptiveTree.trainOnInstanceImpl(trainInstance);
+    }
+
+    /**
+     * @param modelEvaluation Prequential Model Evaluator.
+     * @param cepEvent        event data
+     * @param classValue      class label of the cepEvent
+     * @return Prequential accuracy
+     */
+    public double evaluationTrainOnEvent(PrequentialModelEvaluation modelEvaluation,
+                                         double[] cepEvent, String classValue) {
+        int classIndex = cepEvent.length - 1;
+
+        //create instance with only the feature attributes
+        double[] test = Arrays.copyOfRange(cepEvent, 0, classIndex);
+        Instance testInstance = createMOAInstance(test);
+
+        double[] votes = hoeffdingAdaptiveTree.getVotesForInstance(testInstance);
+
+        cepEvent[classIndex] = getClasses().indexOf(classValue);
+        Instance trainInstance = createMOAInstance(cepEvent);
+
+        hoeffdingAdaptiveTree.trainOnInstanceImpl(trainInstance);
+        modelEvaluation.addResult(trainInstance, votes);
+        return MathUtil.roundOff(modelEvaluation.getFractionCorrectlyClassified(), 3);
+    }
+
+    /**
+     * @param cepEvent Event data.
+     * @return predicted class index, probability of the prediction.
+     */
+    public Object[] getPrediction(double[] cepEvent) {
+        Instance testInstance = createMOAInstance(cepEvent);
+        double[] votes = hoeffdingAdaptiveTree.getVotesForInstance(testInstance);
+        int classIndex = CoreUtils.argMaxIndex(votes);
+        double confidenceLevel = getPredictionConfidence(votes);
+        return new Object[]{classIndex, confidenceLevel};
+    }
+
+    /**
+     * @param cepEvent Event Data
+     * @return represents a single Event
+     */
+    private Instance createMOAInstance(double[] cepEvent) {
+        Instance instance = new DenseInstance(1.0D, cepEvent);
+        //set schema header for the instance
+        instance.setDataset(streamHeader);
+        return instance;
+    }
+
```

Comment:
If we are having java doc comments , then apply for every method for consistency